nifi-nar-bundles/nifi-standard-services/nifi-hbase-client-service-api/src/main/java/org/apache/nifi/hbase/HBaseClientService.java

```diff
@@ -196,4 +196,15 @@
      */
     byte[] toBytesBinary(String s);
 
+    /**
+     * Create a transit URI from the current configuration and the specified table name.
+     * The default implementation just prepend "hbase://" to the table name and row key, i.e. "hbase://tableName/rowKey".
+     * @param tableName The name of a HBase table
+     * @param rowKey The target HBase row key, this can be null or empty string if the operation is not targeted to a specific row
+     * @return a qualified transit URI which can identify a HBase table row in a HBase cluster
+     */
+    default String toTransitUri(String tableName, String rowKey) {
+        return "hbase://" + tableName + (rowKey != null && !rowKey.isEmpty() ? "/" + rowKey : "");
```

Comment:
Are there any scenarios where `rowKey` will be null or empty?

Reply:
Currently probably no scenarios, however if NiFi starts providing operations against HBase tables meaning not with a specific row, such as DDL, Phenix query or passing scan result as Record then there will be situations rowKey is null.